### DISCUSSION TITLE
two things...

### DISCUSSION THREAD

**Discussion by David-ksturnpike** [Wakoma/nimble#73] [General]
massive props for whomever thought of and created the assembly guide

for the assembly guide is there a way to "blow up" the pictures so that we can see what some of the items look like? specifically, the mounting of the wireless APs in the lid?  

second question how do I "test" the assembly guide?

thank you

**Comment by jmwright** [comment]
@David-ksturnpike Exploded views are in-progress: https://github.com/Wakoma/nimble/discussions/61

The assembly guide is created using GitBuilding, and @julianstirling can explain more about that.

By "test" do you mean build a Nimble with the guide, or auto-generate an assembly guide for a given configuration?

**Reply by julianstirling** [reply]
Hi @David-ksturnpike.

There is a full-ish example guide for a specific nimble config [here](https://julianstirling.github.io/nimble-example-documentation/). This was put together with some pictures that were available at the time. The code for it is here in this [PR](https://github.com/Wakoma/nimble-example-documentation/pull/1).

The plan for this example guide was to be a bit of a template for what we want the automatically generated guide [here](https://wakoma.github.io/nimble/assembly-docs/) to look like. Jeremy is working on getting the exploded views to render. The plan is to make the assembly documentation work something like we did in the past with [OpenFlexure](https://build.openflexure.org/openflexure-microscope/v7.0.0-beta2/high_res_optics_module.html). As Jeremy said, all the docs are using a program called [GitBuilding](https://gitbuilding.io), due to the modular nature of the docs we are using an alpha release with a number of new features I still need to document.

Hopefully soon we will have a server hosted somewhere so that the assembly guide will be generated for any arbitrary config. Over time we will add the features to get all of the exploded renders in.

Long term it would be really cool to do renders that explode as a slider is activated.. like the 3rd example on this page:
https://gitbuilding.gitlab.io/gb3d/

**Reply by David-ksturnpike** [reply]
Hi @julianstirling,

i looked through your "[full-ish guide](https://julianstirling.github.io/nimble-example-documentation/)" and I found that it was very very helpful, even in its current state.  With that being said i do have some suggestions for your guide.  I saw that you had a lot of pictures of the rack and how to setup and put together the rack (which is very good) however I did not see any pictures of how to put together the mounting bracket for the three AP's in the lid, just a very small sliver of the finished product.  while that may be ok, it doesn't "flow" with the rest of the setup/site.     Second suggestion would be to take some pictures of the shelves with nothing in them, that way if you have to reference that particular shelf then you have a picture of it rather than just sending the person to the page where the STL's are located.  consequently, if they find that they don't have that particular shelf already printed then on the page with the picture have a link to the stl of that piece so that they can print it.   this isn't a suggestion more of a question, when you build the rack could you build it from the "bottom up" that being said that you would put the bottom plate and then attach the four posts, after that adding the shelves one on top of each other connecting them to the posts as you go up the stack. the other question, seeing that there is a four-post rack why we aren't using all four posts when creating the rack structure? i know that someone else created a [shelf](https://www.printables.com/model/348744-tp-link-tl-sg105-holder-for-wakoma-nimble-rack-sys) that is using all four posts. could we incorporate that into the structure?

thoughts?

**Comment by julianstirling** [comment]
Great points David. In answer to the questions on why are there not shots of certain steps, or certain angles, or in certain assembly states. The answer is that the guide was made by me by screenshotting some videos that were available. So yes there were shots I would have added but they were not in the video..

Soon we will have renders for these steps. So once we build the rendered guide we should be able to add in extra renders that are needed.

The order we add the shelves is a good open question. Again we were repeating an example assembly guide. We may well move towards top to bottom assembly.  I can see that it simplifies some things.

Finally I think you are asking why the shelves don't attach to both the front and back legs. I think this is generally due to the ease of replacing shelves only from the front (except the few broad ones that need adding from top). It is the standard way that server racks are constructed. I think if we have shelves that are very heavy that sag then attaching to the back legs is certainly something to consider.
